node.go

```diff
@@ -130,28 +142,24 @@ func (node *Node) Visit(opts *Options) (dirs, files int) {
 			vpaths: node.vpaths,
 		}
 		d, f := nnode.Visit(opts)
+		// "prune" option, hide empty directories
+		if nnode.IsDir() && opts.Prune && f == 0 {
+			continue
+		} else if nnode.IsDir() && opts.MatchDirs && opts.IPattern != "" && nnode.match(opts.IPattern, opts) {
+			continue
+		}
 		if nnode.err == nil && !nnode.IsDir() {
```

Comment:
> Regarding this line:
> 
> > If the --prune option is used, empty folders that match the pattern will not be pruned.
> 
> Don't we need to test this instead?
> 
> ```go
> if opts.Prune && f == 0 && (opts.Pattern == "" || !nnode.match(opts.Pattern, opts)) {
> ```

I am fine with the suggested code snippet, but this comment is mixing cases so .. use it instead of what?